src/foam/nanos/auth/ThemeAuthService.js

```diff
@@ -66,41 +55,27 @@ foam.CLASS({
         }
       ],
       javaCode: `
-        User user = getUser(x, identifier);
-        if ( user == null ) {
-          throw new AuthenticationException("User not found.");
-        }
-        return super.login(x, identifier, password);
-      `
-    },
-    {
-      name: 'getUser',
-      documentation: 'Convenience method to get a user by username or email',
-      type: 'User',
-      args: [
-        {
-          name: 'x',
-          type: 'Context'
-        },
-        {
-          name: 'identifier',
-          type: 'String'
+        AuthService auth = (AuthService) x.get("auth");
+        User user = super.login(x, identifier, password);
+
+        // Get theme's spid and user's spid
+        Theme theme = ((Theme) x.get("theme"));
+        String themeSpid = theme.getSpid();
+        String userSpid = user.getSpid();
+
+        // Check if theme's spid and user's spid matched.
+        // if matched pass, else throw an error.
+        // e.g., throws error: userSpid: "treviso", themeSpid: "intuit"
+        if (
+          user != null &&
+          ! userSpid.equals(themeSpid) &&
+          ! userSpid.equals(getSuperSpid())
```

Comment:
we should also check if superSpid is not an empty string here or else we will allow users with a spid to pass this if superSpid is not defined.

Reply:
okay, thank you!